### PR TITLE
[Darkside] Added development environment for new CSS

### PR DESCRIPTION
**Pull request by KenAJoh**
### Storybook
Can now toggle between old (default) and darkside (new) CSS in storybook. Can also change between light and darkmode with an updated toggle

Screenshot 2024-10-14 at 14 23 54

### Development

[README.md](https://github.com/navikt/aksel/blob/b87489ad6f1ed98afbd64b440b490a0f95fffd0c/%40navikt/core/css/darkside/README.md)

Added new environment to write and test new CSS using new tokens under `core/css/darkside`
- Used by storybook when `mode` is set to darkside (with HMR support)
- Used by "referansesider"
- Not included in css-build, so can work in isolation

### Component Checklist 📝

- [x] JSDoc
- [x] Examples
- [x] Documentation
- [x] Storybook
- [x] Style mappings (`@navikt/core/css/config/_mappings.js`)
- [x] Component tokens (`@navikt/core/css/tokens.json`)
- [x] CSS class deprecations (`@navikt/aksel-stylelint/src/deprecations.ts`)
- [x] Exports (`@navikt/core/react/src/index.ts` and `@navikt/core/react/package.json`)
- [x] New component? CSS import (`@navikt/core/css/index.css`)
- [x] Breaking change? Update migration guide. Consider codemod.
- [x] Changeset (Format: `<Component>: <gitmoji?> <Text>.` E.g. "Button: :sparkles: Add feature xyz.")

### CODE REVIEW

File: .storybook/PreviewDarkside.jsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    document.body.style.setProperty("background", "var(--a-bg-default)");
+    return () => {
+      document.head.removeChild(style);
+      document.body.style.removeProperty("background");
+    };
+  }, []);
+
+  return (
+    <div style={{ background: "var(--a-bg-default) !important" }}>
```
Comment (HalvorHaugan) [inline .storybook/PreviewDarkside.jsx]: Is this needed?

File: .storybook/PreviewDefault.jsx
Comment (HalvorHaugan) [inline .storybook/PreviewDefault.jsx:1]: Consider merging PreviewDefault and PreviewDarkside

File: .storybook/preview.jsx [hunk 2]
```diff
@@ -16,44 +18,52 @@ export const parameters = {
...
+          <DarkSideDekorator>
+            <StoryFn />
+          </DarkSideDekorator>
+        ) : (
+          <DefaultDekorator>
+            <StoryFn />
+          </DefaultDekorator>
+        )}
+      </div>
+    );
```
Comment (HalvorHaugan) [inline .storybook/preview.jsx]:
```suggestion
    return context.globals.mode === "darkside" ? (
      <DarkSideDekorator>
        <StoryFn />
      </DarkSideDekorator>
    ) : (
      <DefaultDekorator>
        <StoryFn />
      </DefaultDekorator>
    );
```

File: @navikt/core/css/darkside/baseline/index.css [hunk 3]
```diff
@@ -0,0 +1,27 @@
+@import "./fonts.css";
+
+/**
+ * Baseline
+ * Missing from "original"
+ * - CSS-resets
+ * - Print styles
+ * - utilities (sr-only, animation-reset etc)
+ * - Tokens for: font, font-size, line-height, shadows etc
```
Comment (HalvorHaugan) [inline @navikt/core/css/darkside/baseline/index.css:9]: Won't we need those tokens? 🤔

File: .storybook/PreviewDarkside.jsx (hunk 1 above)
Comment (KenAJoh) [reply .storybook/PreviewDarkside.jsx]: Nope, removed it

File: @navikt/core/css/darkside/baseline/index.css (hunk 3 above)
Comment (KenAJoh) [reply @navikt/core/css/darkside/baseline/index.css:9]: Yes, but can be added in separate PR(s) 🚀

File: .storybook/preview.jsx [hunk 4]
```diff
@@ -16,44 +18,70 @@ export const parameters = {
...
+    }
+
+    document.body.style.setProperty("background", "var(--a-bg-default)");
+    return () => {
+      document.body.style.removeProperty("background");
+    };
+  }, [mode]);
+
+  return children;
+};
```
Comment (HalvorHaugan) [inline .storybook/preview.jsx]: What about this?
```suggestion
    if (mode === "darkside") {
      document.body.style.setProperty("background", "var(--a-bg-default)");
    }

    return () => {
      document.head.removeChild(style);
      document.body.style.removeProperty("background");
    };
  }, [mode]);
```

File: .storybook/preview.jsx [hunk 5]
```diff
@@ -43,6 +43,32 @@ export const initialGlobals = {
   mode: "default",
 };
 
+const ThemeDecorator = ({ children, mode }) => {
```
Comment (HalvorHaugan) [inline .storybook/preview.jsx]: Should we call it `ModeDecorator`, to not confuse it with theme?

**Review by HalvorHaugan — DISMISSED** [review]

**Review by JulianNymark — APPROVED** [review]
🚀

**Comment by JulianNymark** [comment]
> Found potential bug

what was the potential bug 😳 (I hope it was false potential! 🙏 )

**Comment by HalvorHaugan** [comment]
> > Found potential bug
> 
> what was the potential bug 😳 (I hope it was false potential! 🙏 )

It looked like some tokens wasn't loaded, but it seems to only affect "Getting started":
image

**Comment by KenAJoh** [comment]
> > > Found potential bug
> > 
> > 
> > what was the potential bug 😳 (I hope it was false potential! 🙏 )
> 
> It looked like some tokens wasn't loaded, but it seems to only affect "Getting started": image

Fixed now: https://github.com/navikt/aksel/commit/8b373e8b80b4b51a60d5ab6915e24b41ba738682

**Pull request merged**